pkg/kvstore/client.go

```diff
@@ -23,17 +23,17 @@ var (
 
 func initClient(module backendModule) error {
 	c, errChan := module.newClient()
-
-	defaultClient = c
-
-	go func() {
-		err, isErr := <-errChan
+	select {
+	case err, isErr := <-errChan:
```

Comment:
Looks fine, but any concerns about making this synchronous vs. asynchronous with respect to timing operations , especially in relation to the etcd-operator?

Reply:
It's still async due to the `select { default: }`. I'm actually a bit more worried about what the current behavior is of suddenly exiting with the fatal message. I'll read up on etcd docs when this happens exactly.

Reply:
@ianvernon concerns are relevant here. This is modifying how we "Fatal" in case etcd-client cannot reach etcd-server. By patching the code as it is right now I've a feeling the the cilium-agent will never exit by fatal itself if it can not reach etcd-server.

Reply:
We can do something like this:
```
--- a/pkg/kvstore/client.go
+++ b/pkg/kvstore/client.go
@@ -23,6 +23,10 @@ var (

 func initClient(module backendModule) error {
        c, errChan := module.newClient()
+       if c == nil {
+               err := <-errChan
+               log.WithError(err).Fatalf("Unable to connect to kvstore")
+       }

        defaultClient = c
```